File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/encryption/EncryptionFragment.java
package gateway.hitrontech.com.encryption.fragment.encryption;

import android.annotation.SuppressLint;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.jakewharton.rxbinding2.view.RxView;
import com.kryst.njit.base.BaseFragment;
import gateway.hitrontech.com.encryption.R;
import gateway.hitrontech.com.encryption.databinding.FragmentEncryptionBinding;
import gateway.hitrontech.com.encryption.utils.SharePreManager;
import io.reactivex.functions.Consumer;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class EncryptionFragment extends BaseFragment implements Contract.View {

  private FragmentEncryptionBinding mBinding;
  private Contract.Presenter mPresenter;

  public static Fragment getInstance() {
    Bundle args = new Bundle();
    Fragment fragment = new EncryptionFragment();
    fragment.setArguments(args);
    return fragment;
  }

  @NotNull
  @Override
  protected View createView(@NotNull LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {

    if (!getViewExisted()) {
      mBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_encryption, container, false);

    }
    return mBinding.getRoot();
  }

  @SuppressLint("CheckResult")
  @Override
  protected void viewFound(@Nullable View view) {
    if (!getViewExisted()) {
      new Presenter(this);
      mPresenter.start();

      RxView.clicks(mBinding.encryption)
          .subscribe(new Consumer<Object>() {
            @Override
            public void accept(Object click) throws Exception {
              mPresenter.getEncryptionText(mBinding.plainText.getText().toString());
            }
          });

      RxView.clicks(mBinding.changePlainText)
          .subscribe(new Consumer<Object>() {
            @Override
            public void accept(Object o) throws Exception {
              if (mBinding.plainText.getText().toString().equals(getString(R.string.test_number))) {
                mBinding.plainText.setText(getString(R.string.ht_security_key));
              } else {
                mBinding.plainText.setText(getString(R.string.test_number));
              }
            }
          });

    }
  }

  @NotNull
  @Override
  protected String getTitle() {
    return "加密";
  }

  @Override
  public void setEncryptionText(String cipherText) {
    SharePreManager.getInstance().setCipherText(cipherText);
    mBinding.cipherText.setText(cipherText);
  }

  @Override
  public void setPresenter(@NonNull Contract.Presenter presenter) {
    this.mPresenter = presenter;
  }
}

File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/encryption/EncryptionFragment.kt
package gateway.hitrontech.com.encryption.fragment.encryption

import android.annotation.SuppressLint
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.jakewharton.rxbinding2.view.RxView
import gateway.hitrontech.com.encryption.R
import gateway.hitrontech.com.encryption.base.BaseFragment
import gateway.hitrontech.com.encryption.databinding.FragmentEncryptionBinding
import gateway.hitrontech.com.encryption.utils.SharePreManager

class EncryptionFragment : BaseFragment(), Contract.View {

    private var mBinding: FragmentEncryptionBinding? = null
    private var mPresenter: Contract.Presenter? = null

    override fun createView(inflater: LayoutInflater, container: ViewGroup?,
                            savedInstanceState: Bundle?): View {

        if (!viewExisted) {
            mBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_encryption, container, false)

        }
        return mBinding!!.root
    }

    @SuppressLint("CheckResult")
    override fun viewFound(view: View?) {
        if (!viewExisted) {
            Presenter(this)
            mPresenter!!.start()

            RxView.clicks(mBinding!!.encryption)
                    .subscribe { mPresenter!!.getEncryptionText(mBinding!!.plainText.text!!.toString()) }

            RxView.clicks(mBinding!!.changePlainText)
                    .subscribe {
                        if (mBinding!!.plainText.text!!.toString() == getString(R.string.test_number)) {
                            mBinding!!.plainText.setText(getString(R.string.ht_security_key))
                        } else {
                            mBinding!!.plainText.setText(getString(R.string.test_number))
                        }
                    }

        }
    }

    override fun getTitle(): String {
        return "加密"
    }

    override fun setEncryptionText(cipherText: String) {
        SharePreManager.instance.cipherText = cipherText
        mBinding!!.cipherText.setText(cipherText)
    }

    override fun setPresenter(presenter: Contract.Presenter) {
        this.mPresenter = presenter
    }

    companion object {

        val instance: Fragment
            get() {
                val args = Bundle()
                val fragment = EncryptionFragment()
                fragment.arguments = args
                return fragment
            }
    }
}
